### PR TITLE
Create logic for displaying toggle in modals

### PR DESCRIPTION
**Pull request by alicjakujawa**
## Description

This PR adds in logic for displaying toggle in modals

**New stuff** ✨

* Added hook which return information if extension is enabled
* Fetching and passing this flag in ColonyHomeActions and ColonyFundingMenu

**Changes** 🏗

* Updated heading to lets us add toggle inline with title
* Updated core Toggle component to support Message descriptors

Resolves DEV-250

### CODE REVIEW

File: src/modules/core/components/Fields/Toggle/Toggle.tsx [hunk 1]
```diff
@@ -18,7 +19,7 @@ interface Appearance {
 interface Props {
   appearance?: Appearance;
   name: string;
-  label?: string;
+  label?: string | MessageDescriptor;
```
Comment (rdig) [inline src/modules/core/components/Fields/Toggle/Toggle.tsx:22]: This should also have a `labelValues` which will bring in the (potentially) message descriptor variables

File: src/modules/core/components/Fields/Toggle/Toggle.tsx [hunk 2]
```diff
@@ -18,7 +19,7 @@ interface Appearance {
...
   disabled?: boolean;
   tooltipText?: string;
```
Comment (rdig) [inline src/modules/core/components/Fields/Toggle/Toggle.tsx:24]: Same for tooltip text, should be a message descriptor and there should also be a `tooltipTextValues` prop.

_(Yes, I know you didn't add them like this, but while you are at it, might as well make the change)_

File: src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx [hunk 3]
```diff
@@ -65,6 +67,7 @@ const ColonyFundingMenu = ({
   const handleMintTokens = useCallback(() => {
     openTokenMintDialog({
       colony,
+      isVotingExtensionEnabled,
     });
   }, [colony, openTokenMintDialog]);
```
Comment (rdig) [inline src/modules/dashboard/components/ColonyFundingMenu/ColonyFundingMenu.tsx:72]: This needs `isVotingExtensionEnabled` as a dependency, no ?

File: src/modules/dashboard/components/ManageFundsDialog/ManageFundsDialog.tsx [hunk 4]
```diff
@@ -106,14 +107,17 @@ const ManageFundsDialog = ({
...
   const { walletAddress, username, ethereal } = useLoggedInUser();
 
   const allUserRoles = useTransformer(getAllUserRoles, [colony, walletAddress]);
 
   const hasRegisteredProfile = !!username && !ethereal;
   const canMoveFunds = hasRegisteredProfile && canFund(allUserRoles);
-  const canMintNativeToken = colony.canMintNativeToken && hasRoot(allUserRoles);
+  const canMintNativeToken =
+    (colony.canMintNativeToken && hasRoot(allUserRoles)) ||
+    isVotingExtensionEnabled;
```
Comment (rdig) [inline src/modules/dashboard/components/ManageFundsDialog/ManageFundsDialog.tsx:120]: While the logic is good, I think this is hard to read.

I would have left the `canMintNativeToken` logic as it is, and just change the check in the object:
```
    {
      title: MSG.mintTokensTitle,
      description: MSG.mintTokensDescription,
      icon: 'emoji-seed-sprout',
      permissionRequired: !isVotingExtensionEnabled || !canMintNativeToken,
      permissionInfoText: MSG.permissionsListText,
      permissionInfoTextValues: {
        permissionsList: (
          <FormattedMessage {...MSG.mintTokensPermissionsList} />
        ),
      },
      onClick: () => callStep(nextStepMintTokens),
    },
```

File: src/modules/dashboard/components/TokenMintDialog/TokenMintForm.tsx [hunk 5]
```diff
@@ -70,8 +77,11 @@ const TokenMintForm = ({
           appearance={{ size: 'medium', margin: 'none', theme: 'dark' }}
           text={MSG.title}
         />
+        {userHasPermissions && isVotingExtensionEnabled && (
+          <Toggle label={MSG.force} name="toggle" />
```
Comment (rdig) [inline src/modules/dashboard/components/TokenMintDialog/TokenMintForm.tsx:81]: We need a proper name for this. How about `forceMotion`

File: src/modules/core/components/Dialog/DialogSection.css [hunk 6]
```diff
@@ -5,6 +5,9 @@
 }
 
 .themeHeading {
+  display: flex;
+  justify-content: space-between;
+  align-items: center;
```
Comment (rdig) [inline src/modules/core/components/Dialog/DialogSection.css:10]: Please check if this change doesn't break the styles of the modals that don't a toggle.

**Review by rdig — APPROVED** [review]
Looks good to me. I have left a few comments that need to be addressed, but after that, you can merge it.

**Review by ArmandoGraterol — APPROVED** [review]
Nice! So nice I'll show this PR to my mom :+1:

File: src/modules/core/components/Dialog/DialogSection.css (hunk 6 above)
Comment (alicjakujawa) [reply src/modules/core/components/Dialog/DialogSection.css:10]: i was checking :)

**Pull request merged**